_posts/en/newsletters/2024-12-13-newsletter.md

```diff
@@ -0,0 +1,462 @@
+answers.  Click on a question below to see a summary of the answer from
+the meeting.*
+
+[Track and use all potential peers for orphan resolution][review club 31397]
+is a PR by [glozow][gh glozow] that improves the reliability of
+orphan resolution by letting the node request missing ancestors from all
+peers instead of just the one that announced the orphan. It does so by
+introducing `m_orphan_resolution_tracker` which is responsible for
+remembering which peers are candidates for orphan resolution and for
+scheduling when to make orphan resolution requests. The approach is
+designed to be bandwidth-efficient, to not be vulnerable to censorship,
+and to load-balance between peers.
+
+{% include functions/details-list.md
+  q0="What is orphan resolution?"
+  a0="A transaction is referred to as an orphan when the node doesn't
+  have at least one of the transactions the orphan is spending from.
+  Orphan resolution is the process of trying to find these missing
+  transactions."
+  a0link="https://bitcoincore.reviews/31397#l-23"
+
+  q1="Prior to this PR, what are the steps for orphan resolution,
+  starting from when we notice that the transaction has missing inputs?"
+  a1="When a node receives an orphan transaction, it will request the
+  parent transaction from the same peer that sent the orphan. Other
+  peers aren't actively queried, but may still send us the parent, for
+  example when they announce it via an INV message, or when they send
+  another orphan with the same missing parent."
+  a1link="https://bitcoincore.reviews/31397#l-29"
+
+  q2="What are the ways we may fail to resolve an orphan with the peer
+  we request its parents from? What are some reasons things this may
+  happen, honest or otherwise?"
+  a2="An honest peer may simply have been disconnected, or it may have
+  evicted the parent from its mempool. A malicious peer may just not
+  respond to the request, or it may send a parent with malleated,
+  invalid witness data which results in the parent having the expected
+  txid while failing validation."
+  a2link="https://bitcoincore.reviews/31397#l-49"
+
+  q3="How could an attacker prevent a node from downloading a 1p1c
+  package by exploiting today’s orphan resolution behavior?"
+  a3="An attacker can announce a malleated orphan (see Q2) unsolicited.
+  Once the malleated orphan transaction is accepted in the orphanage,
+  the honest orphan transaction will no longer be accepted because it
+  has the same txid. This prevents the package from being relayed.
+  Alternatively, an attacker could flood a node with orphan
+  transactions. Because the orphanage is limited in size, and
+  transactions are evicted randomly, this will affect how well a node
+  can download 1p1c packages."
+  a3link="https://bitcoincore.reviews/31397#l-64"
+
+  q4="What is the PR’s solution to the problem in the previous question?"
+  a4="Instead of adding the missing parent transaction to the
```

Comment:
might help convey that the parent is the thing that is missing

```suggestion
  a4="Instead of adding the missing-parent transaction to the
```

Reply:
I think @stickies-v version without the hyphen is correct here, so I made a different edit to hopefully clarify.